### PR TITLE
Automatic TCU

### PR DESCRIPTION
**Pull request by Nyconing**
In the script.cpp section, the codes actually remains unchanged, but github file comparer mess it up.

### CODE REVIEW

File: Gears/ScriptSettings.cpp [hunk 1]
```diff
@@ -549,6 +551,7 @@ void ScriptSettings::parseSettingsGeneral() {
     AutoParams.CurrGearMinRPM = ini.GetDoubleValue("AUTO_PARAMS", "CurrGearMinRPM", AutoParams.CurrGearMinRPM);
     AutoParams.EcoRate = ini.GetDoubleValue("AUTO_PARAMS", "EcoRate", AutoParams.EcoRate);
     AutoParams.DownshiftTimeoutMult = ini.GetDoubleValue("AUTO_PARAMS", "DownshiftTimeoutMult", AutoParams.DownshiftTimeoutMult);
+    AutoParams.UsingATCU = ini.GetDoubleValue("AUTO_PARAMS", "UsingATCU", AutoParams.DownshiftTimeoutMult);
```
Comment (E66666666) [inline Gears/ScriptSettings.cpp:554]:
```suggestion
    AutoParams.UsingATCU = ini.GetBoolValue("AUTO_PARAMS", "UsingATCU", AutoParams.UsingATCU);
```

File: Gears/VehicleConfig.cpp [hunk 2]
```diff
@@ -74,6 +74,7 @@ void VehicleConfig::loadSettings(const ScriptSettings& gSettings, const std::str
     AutoParams.CurrGearMinRPM = ini.GetDoubleValue("AUTO_PARAMS", "CurrGearMinRPM", gSettings.AutoParams.CurrGearMinRPM);
     AutoParams.EcoRate = ini.GetDoubleValue("AUTO_PARAMS", "EcoRate", gSettings.AutoParams.EcoRate);
     AutoParams.DownshiftTimeoutMult = ini.GetDoubleValue("AUTO_PARAMS", "DownshiftTimeoutMult", gSettings.AutoParams.DownshiftTimeoutMult);
+    AutoParams.UsingATCU = ini.GetDoubleValue("AUTO_PARAMS", "UsingATCU", gSettings.AutoParams.UsingATCU);
```
Comment (E66666666) [inline Gears/VehicleConfig.cpp:77]:
```suggestion
    AutoParams.UsingATCU = ini.GetBoolValue("AUTO_PARAMS", "UsingATCU", gSettings.AutoParams.UsingATCU);
```

File: Gears/VehicleData.hpp [hunk 3]
```diff
@@ -72,6 +73,9 @@ struct VehicleGearboxStates {
     float EngineLoad = 0.0f;
     float UpshiftLoad = 0.0f;
     float DownshiftLoad = 0.0f;
+
+    //ATCU
+    AtcuGearbox Atcu = AtcuGearbox();
```
Comment (E66666666) [inline Gears/VehicleData.hpp:78]: Didn't mean it like that, but I guess this is fine.

File: Gears/script.cpp [hunk 4]
```diff
@@ -104,7 +106,7 @@ void updateSteeringMultiplier();
 //                           Mod functions: Shifting
 ///////////////////////////////////////////////////////////////////////////////
 
-void shiftTo(int gear, bool autoClutch);
+//void shiftTo(int gear, bool autoClutch); moved to h
```
Comment (E66666666) [inline Gears/script.cpp:109]: Comment can be removed.

File: Gears/script.cpp [hunk 5]
```diff
@@ -1139,60 +1141,70 @@ void functionAShift() {
...
-        nextGearMinSpeed = g_settings().AutoParams.NextGearMinRPM * g_vehData.mDriveMaxFlatVel / g_vehData.mGearRatios[currGear + 1];
-    }
+    bool skidding = isSkidding(3.5f);
 
-    float currGearMinSpeed = g_settings().AutoParams.CurrGearMinRPM * g_vehData.mDriveMaxFlatVel / g_vehData.mGearRatios[currGear];
+    if (g_settings().AutoParams.UsingATCU) {
+        AtcuLogic::Cycle();
+    }
+    else {
+        float nextGearMinSpeed = 0.0f; // don't care about top gear
```
Comment (E66666666) [inline Gears/script.cpp:1150]: Additional indenting for the `else` case is a bit annoying. You don't have to do anything about this now, but you might just want to do your stuff and then early-return after handling your stuff

Comment (Nyconing) [reply Gears/script.cpp:1150]: not sure about this, you mean
`} else {` ?

Comment (Nyconing) [reply Gears/script.cpp:1149]: those formatting styles are given by my VS2019 default... did VS having feature similar like lint? We may use it so the code styles can be unified.

**Comment by E66666666** [comment]
I'll merge it and touch up some things.

**Pull request merged**